Update plugin to support multiple plugins in the same project

mic345 opened the pull request:
In short this pull request includes:

1. Support multiple plugins in the same project
2. Some refactoring on the web plugin to simplify the code and prepare for code reuse among all firebase plugins.

Commit:
fixed compilation error TS2794: Expected 1 arguments, but got 0

Commit:
Added initial support for multi Firebase plugins in the same project

Commit:
fixed Firebase script is loaded several times

Commit:
[refactor] gather plugins' common code together

Commit:
[refactor] unify plugin's common code with capacitor-community/firebase-remote-config

Commit:
fixed firebase may not be ready on firebaseObjectReadyPromise

markxoe commented:
That's a very useful feature 👍🏻

Commit:
[privacy] removed unrequired Firebase/Core pod

markxoe reviewed (approved):
I'd love that to be in the next Version

brownoxford commented:
Hi @mic345 Thanks for the PR! I'd prefer to see this broken into several PRs, e.g., one for TS2794, one for the script loading issue, and one for your new functionality (I've added some more comments to the pr regarding these). Can you also add a description of the use case that you are solving for?

brownoxford commented on src/web.ts:
```diff
@@ -289,15 +289,15 @@ export class FirebaseAnalyticsWeb extends WebPlugin
         document.getElementById(scripts[1])
       ) {
-        return resolve();
+        return resolve( null );
```
Instead of passing a value where none is expected, can you modify the promise to declare a type, _e.g._:
```
return new Promise<void>(async (resolve, _reject) => {
```

brownoxford commented on src/web.ts:
```diff
@@ -273,26 +273,8 @@ export class FirebaseAnalyticsWeb extends WebPlugin
   /**
    * Check for existing loaded script and load new scripts
    */
-  private loadScripts() {
```
When you split this out into a separate PR, can you add some details about what is going wrong and what you did to fix?

mic345 replied:
> can you add some details about what is going wrong and what you did to fix?

If I remember correctly, we did this in order to create a reusable code between `firebase-analytics` and `firebase-remote-config` for their common needs (loading the scripts, etc.).

In an ideal world someone would create a `firebase-common` project, move all the common code there and would use that project both `firebase-analytics`, `firebase-remote-config` and anywhere else firebase needs to load scripts (likely everywhere). Over time more common code would move there.

> When you split this out into a separate PR, 

I apologize for the single PR. We issue those fixes in the same PR as we needed to proceed with our work and the original project was discarded. Different PRs would have never been merged on time.

You are rightfully asking for splitting, but I'm afraid I wont have time for it anytime soon, as I'm already over loaded with work 😬. At this point I can suggest either to create a branch on the original project, merge the whole PR to it and test/refactor as much as needed, and only than merge to the master, or to cherry pick the commits from the PR. If it was me, I'd go for the first option, as the entire PR is already tested and used on our project.

I hope you find our code helpful.

Commit:
fixed [bug] project supports capacitor 2

Commit:
update depentant packages

Commit:
fixed [bug] method setScreenName is no longer supported.

For alternative methods see https://firebase.googleblog.com/2020/08/google-analytics-manual-screen-view.html
